src/main/scala/analysis/InterprocSteensgaardAnalysis.scala

```diff
@@ -245,98 +242,130 @@ class InterprocSteensgaardAnalysis(
     }
   }
 
+  def memLoadToRegion(memLoad: MemoryLoad, cmd: Command): Set[MemoryRegion] = {
+    if (mmm.getStack(cmd).nonEmpty) {
+      mmm.getStack(cmd).asInstanceOf[Set[MemoryRegion]]
+    } else {
+      val isGlobal = evaluateExpression(memLoad.index, constantProp(cmd))
+      if (isGlobal.isDefined) {
+        val globalRegion = mmm.findDataObject(isGlobal.get.value)
+        if (globalRegion.isDefined) {
+          return Set(globalRegion.get)
+        }
+        return Set.empty[MemoryRegion] // TODO: IT SHOULD THROW AN EXCEPTION
+        //throw Exception(s"Could not find region for MemLoad: $memLoad, Command: $cmd, Eval: $isGlobal, Global: $globalRegion")
+      }
+      memLoad.index match // treats case where the index is a region and is loaded again like in jumptable2/clang_pic
+        case variable: Variable =>
+          val ctx = getUse(variable, cmd, reachingDefs)
+          for (i <- ctx) {
+            i.rhs match {
+              case load: MemoryLoad =>
+                return memLoadToRegion(load, i)
```

Comment:
Having a return inside a for loop like this is deprecated in Scala and gives a warning. Since getUse returns a set, why do we only return a single result here?